test/parallel/test-stream-duplex-readable-end.js

```diff
@@ -0,0 +1,30 @@
+'use strict';
+// https://github.com/nodejs/node/issues/35926
+const common = require('../common');
+const stream = require('stream');
+
+let loops = 3;
+
+const src = new stream.Readable({
+  read: common.mustCallAtMost(() => {
+    if (loops--)
+      src.push(Buffer.alloc(20000));
+  }, 2)
```

Comment:
I think this should be a `mustCall()`, not `mustCallAtMost()`. The algorithm is deterministic.